### PR TITLE
Better support for status and initial support for partstat

### PR DESCRIPTION
**Pull request by geier**

### CODE REVIEW

**Comment by madduck** [comment]
I can confirm that this works as intended. Will take discussion back to #580 though.

File: khal/khalendar/event.py [hunk 1]
```diff
@@ -556,6 +568,33 @@ def _alarm_str(self) -> str:
             alarmstr = ''
         return alarmstr
 
+    @property
+    def _status_str(self) -> str:
+        if self.status == 'CANCELLED':
+            statusstr = ' ' + self.symbol_strings['cancelled']
```
Comment (madduck) [inline khal/khalendar/event.py:574]: I'd like to make a case against adding a leading space, or at least to make this configurable. Such a leading space should really be part of the format specification, though I realise it's currently not possible to work with conditionals there.

However, `{status: <1}` will probably suit anyone better when the status symbol is to be used mid-field, whereas adding all the statuses to the end of the line won't make a difference anyway.

Comment (geier) [reply khal/khalendar/event.py:572]: fixed

**Comment by geier** [comment]
config file now takes several addresses per account, but we expect that only one will match.

**Comment by geier** [comment]
@WhyNotHugo sorry for bombing you with review requests, feel free to ignore them.

**Comment by geier** [comment]
This PR doesn't cover yet formatting events by their status as it's a bit more difficult to get this looking the same in khal and ikhal. I'd prefer to merge this without that now, because rebasing against master is becoming tiresome (not especially this branch, but keeping several non-merged features branches around is getting tiring).

**Review by WhyNotHugo — COMMENTED** [review]
If I understand correctly, [the `CALDAV:calendar-user-address-set` property](https://datatracker.ietf.org/doc/html/rfc6638#section-2.4.1) allows querying the address of the owner of a calendar.

This should enable fetching the email address of the calendar owner via vdirsyncer.

Perhaps treating this as calendar metadata might be more future compatible?

**Comment by geier** [comment]
@WhyNotHugo interesting. We should definitively support that, but probably only as a default. At least one of my caldav servers has an email address set that nobody sends email to.

**Pull request merged**